File: mixed-persistence/mixed-persistence-scala-sbt/hello-impl/src/main/scala/com/lightbend/lagom/recipes/hello/impl/HelloServiceImpl.scala
package com.lightbend.lagom.recipes.hello.impl

import com.lightbend.lagom.recipes.hello.api.HelloService
import com.lightbend.lagom.scaladsl.api.ServiceCall
import com.lightbend.lagom.scaladsl.persistence.PersistentEntityRegistry

class HelloServiceImpl(persistentEntityRegistry: PersistentEntityRegistry) extends HelloService {

  override def hello(id: String) = ServiceCall { _ =>
    val ref = persistentEntityRegistry.refFor[HelloEntity](id)
    ref.ask(Hello(id))
  }

  override def useGreeting(id: String) = ServiceCall { request =>
    val ref = persistentEntityRegistry.refFor[HelloEntity](id)
    ref.ask(UseGreetingMessage(request.message))
  }

}

File: i18n/hello-i18n-java-mvn/hello-i18n-impl/src/main/java/com/lightbend/lagom/recipes/i18n/hello/impl/HelloServiceImpl.java
package com.lightbend.lagom.recipes.i18n.hello.impl;

import akka.NotUsed;
import com.lightbend.lagom.javadsl.api.ServiceCall;
import com.lightbend.lagom.recipes.i18n.hello.api.HelloService;

import javax.inject.Inject;
import java.util.concurrent.CompletableFuture;

/**
 * Implementation of the HelloService.
 */
public class HelloServiceImpl implements HelloService {

    @Inject
    public HelloServiceImpl() {
    }

    @Override
    public ServiceCall<NotUsed, String> hello(String id) {
        return request -> CompletableFuture.completedFuture("Hello, " + id + "!");
    }

}
